snap/info.go

```diff
@@ -208,6 +208,17 @@ func ScopedSecurityTag(snapName, scopeName, suffix string) string {
 	return fmt.Sprintf("snap.%s.%s.%s", snapName, scopeName, suffix)
 }
 
+// SplitSecurityTag will extract the snap name from a string of the form
+// `snap.<snap>[.<app>]`. If the tag is not of that form, returns an error.
+func SplitSecurityTag(tag string) (string, error) {
```

Comment:
given that this returns only the name and not all the parts, it should probably be called InstanceNameFromSecurityTag?